internal/state/conditions/conditions.go

```diff
@@ -347,12 +373,15 @@ func NewGatewayAccepted() Condition {
 }
 
 // NewGatewayConflict returns a Condition that indicates the Gateway has a conflict with another Gateway.
-func NewGatewayConflict() Condition {
-	return Condition{
-		Type:    string(v1beta1.GatewayConditionAccepted),
-		Status:  metav1.ConditionFalse,
-		Reason:  string(GatewayReasonGatewayConflict),
-		Message: GatewayMessageGatewayConflict,
+func NewGatewayConflict() []Condition {
+	return []Condition{
+		{
+			Type:    string(v1beta1.GatewayConditionAccepted),
+			Status:  metav1.ConditionFalse,
+			Reason:  string(GatewayReasonGatewayConflict),
+			Message: GatewayMessageGatewayConflict,
+		},
+		NewGatewayNotProgrammedInvalid(GatewayMessageGatewayConflict),
```

Comment:
In this case, should the reason for Programmed also be GatewayReasonGatewayConflict?

Reply:
If we're okay using custom reasons for Programmed condition too.

Reply:
I'm fine with that.  @pleshakov?

Reply:
> In this case, should the reason for Programmed also be GatewayReasonGatewayConflict?

that makes sense

@kate-osborn @sjberman